src/common/procedure/src/local/runner.rs

```diff
@@ -238,11 +238,24 @@ impl Runner {
                 }
                 ProcedureState::Done { .. } => return,
                 ProcedureState::Failed { .. } => return,
+                ProcedureState::Poisoned { .. } => return,
             }
             self.execute_once(ctx).await;
         }
     }
 
+    async fn clean_poisons(&mut self) -> Result<()> {
+        for key in self.meta.poison_keys.iter() {
+            debug!("clean poison: {key}");
+            let key = key.to_string();
+            self.manager_ctx
+                .poison_manager
+                .delete_poison(key, self.meta.id.to_string())
+                .await?;
```

Comment:
Log the keys that failed to delete.

Reply:
resolved in [3331fce](https://github.com/GreptimeTeam/greptimedb/pull/5822/commits/3331fce63408d6f5967551f116194a93890b094c)